MaterialDesignThemes.Wpf/Snackbar.cs

```diff
@@ -1,12 +1,40 @@
 ﻿using System;
+using System.Globalization;
 using System.Windows;
 using System.Windows.Controls;
+using System.Windows.Data;
 using System.Windows.Markup;
 using System.Windows.Media.Animation;
 using System.Windows.Threading;
 
 namespace MaterialDesignThemes.Wpf
 {
+    public enum SnackbarActionButtonPlacementMode
+    {
+        Auto,
+        Inline,
+        SeparateLine
+    }
+
+    public class SnackbarActionButtonPlacementModeConverter : IMultiValueConverter
+    {
+        public object Convert(object[] values, Type targetType, object parameter, CultureInfo culture)
+        {
+            var mode = (SnackbarActionButtonPlacementMode) values[0];
+            var snackbarHeight = (double) values[1];
+            return mode switch
+            {
+                SnackbarActionButtonPlacementMode.Auto when snackbarHeight > 53 => Dock.Bottom,
+                SnackbarActionButtonPlacementMode.Auto => Dock.Right,
+                SnackbarActionButtonPlacementMode.Inline => Dock.Right,
+                SnackbarActionButtonPlacementMode.SeparateLine => Dock.Bottom,
+                _ => Dock.Right
+            };
+        }
+
+        public object[] ConvertBack(object value, Type[] targetTypes, object parameter, CultureInfo culture) => new[] { Binding.DoNothing, Binding.DoNothing };
```

Comment:
In general MultiValueConverters should be only used with mode OneWay, so I am ok with this simply throwing an InvalidOperationException.

Reply:
👍